### PR TITLE
Best practices for libpacemaker node functions

### PR DESCRIPTION
**Pull request by kgaillot**
This continues the project to bring libpacemaker up to current development guidelines, focusing on some node-related functions that move from pcmk_sched_allocate.c to pcmk_sched_nodes.c.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , here's another low-priority review

File: include/crm/common/util.h [hunk 1]
```diff
@@ -49,6 +49,7 @@ int crm_default_remote_port(void);
 int char2score(const char *score);
 char *score2char(int score);
 char *score2char_stack(int score, char *buf, size_t len);
+int pcmk__add_scores(int score1, int score2);
```
Comment (clumens) [inline include/crm/common/util.h:52]: Might as well add unit tests for this function while you're moving it, too.  Looks pretty simple.

File: include/crm/common/util.h [hunk 2]
```diff
@@ -53,9 +58,6 @@ char * crm_strip_trailing_newline(char *str);
 char *crm_strdup_printf(char const *format, ...) G_GNUC_PRINTF(1, 2);
 
 guint crm_parse_interval_spec(const char *input);
-int char2score(const char *score);
-char *score2char(int score);
-char *score2char_stack(int score, char *buf, size_t len);
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply include/crm/common/util.h:58]: Oops, right

File: include/crm/common/util.h (hunk 1 above)
Comment (kgaillot) [reply include/crm/common/util.h:52]: Good idea

**Comment by kgaillot** [comment]
Rebased on current main (no code changes) and added two commits on top for unit tests

**Comment by clumens** [comment]
Looks good to me.

**Pull request merged**